### PR TITLE
check for units attribute in date2index (issue #453)

### PR DESCRIPTION
**Pull request by jswhit**

### CODE REVIEW

**Comment by FrancescAlted** [comment]
LGTM.  Also, you may want to replace this docstring:

"""
If calendar is None, its value is given by nctime.calendar or standard if no such attribute exists.
"""

by

"""
If calendar is None, its value is given by nctime.calendar, raising an error if no such attribute exists.
"""

or similar.

**Comment by jswhit** [comment]
Why?  It does assume 'standard' if there is no ``calendar`` attribute or kwarg.

File: netcdftime/netcdftime.py [hunk 1]
```diff
@@ -1002,6 +1002,10 @@ def date2index(dates, nctime, calendar=None, select='exact'):
     an exact match cannot be found. C{nearest} will return the indices that
     correpond to the closest dates.
     """
+    try: 
+        nctime.units
+    except AttributeError:
+        raise AttributeError("netcdf time variable is missig a 'units' attribute")
```
Comment (FrancescAlted) [inline netcdftime/netcdftime.py]: missig -> missing

File: netcdftime/netcdftime.py [hunk 2]
```diff
@@ -1036,6 +1040,10 @@ def time2index(times, nctime, calendar=None, select='exact'):
     an exact match cannot be found. C{nearest} will return the indices that
     correpond to the closest times.
     """
+    try: 
+        nctime.units
+    except AttributeError:
+        raise AttributeError("netcdf time variable is missig a 'units' attribute")
```
Comment (FrancescAlted) [inline netcdftime/netcdftime.py]: missig -> missing

**Comment by FrancescAlted** [comment]
You are right:  I was reading the docs badly.

**Pull request merged**